File: src/main/scala/temportalist/compression/main/client/GuiConfig.scala
package temportalist.compression.main.client

import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.fml.relauncher.{Side, SideOnly}
import temportalist.compression.main.common.Compression
import temportalist.origin.foundation.client.gui.GuiConfigBase

/**
  *
  * Created by TheTemportalist on 4/14/2016.
  *
  * @author TheTemportalist
  */
@SideOnly(Side.CLIENT)
class GuiConfig(guiScreen: GuiScreen) extends GuiConfigBase(guiScreen, Compression) {}

File: src/main/java/com/temportalist/compression/client/config/GuiConfig.java
package com.temportalist.compression.client.config;

import com.temportalist.compression.common.Compression;
import net.minecraft.client.gui.GuiScreen;
import net.minecraftforge.fml.client.config.IConfigElement;

import java.util.ArrayList;
import java.util.List;

public class GuiConfig extends net.minecraftforge.fml.client.config.GuiConfig {

    public GuiConfig(GuiScreen parentScreen) {
        super(parentScreen, generateConfigElements(),
                Compression.MOD_ID,
                false, false,
                Compression.MOD_NAME
        );
    }

    public static List<IConfigElement> generateConfigElements() {
        List<IConfigElement> elements = new ArrayList<>();
        Compression.main.config.populateConfigElements(elements);
        return elements;
    }

}
